app/forms/support_interface/add_course_form.rb

```diff
@@ -0,0 +1,49 @@
+module SupportInterface
+  class AddCourseForm
+    include ActiveModel::Model
+
+    attr_accessor :course_option_id, :application_form_id, :candidate_id
+
+    validates :course_option_id, presence: true
+    validates :application_form_id, presence: true
+
+    DropdownOption = Struct.new(:id, :name)
+
+    def radio_available_courses
+      @radio_available_courses ||= begin
+        courses.order(:name)
+      end
+    end
+
+    def dropdown_available_courses
+      @dropdown_available_courses ||= courses.map { |course|
+        # YUCK
+
+        course.course_options.map do |course_option|
+          DropdownOption.new(course_option.id, "#{course.name} (#{course.code}) – #{course.provider&.name} - #{course_option.site.name}")
+        end
+      }.flatten.sort_by(&:name)
+    end
+
+    def courses
+      @courses ||= Course.current_cycle.includes(:provider, { course_options: [:site] }).open_on_apply
```

Comment:
We've got 4,810 courses in prod, and rising — does this work ok with that sort of volume?

Reply:
This is only bringing back courses currently open on Apply but I would imagine we may want to rethink the design entirely if the volume grows significantly

Reply:
We would probably want to revert to the design / flow used on candidate side for adding a course but question is whether we do this later or now? @davidgisbey @paulrobertlloyd

Reply:
All those 4,810 are open on Apply: https://www.apply-for-teacher-training.service.gov.uk/support/performance/course-stats, and I guess that's multiplied by locations here. We'll need to choose study mode too.

Reply:
Yea I think this was a potential a reason for the idea of inputting a course code and having a list of available course options presented on a new page. Also +1 to the idea of a confirmation page as i hadn't really thought through the implication of incorrectly sending emails to providers.

Reply:
The idea of not rebuilding the whole course flow is it's quite a bit of work to add a whole new set of controllers and views. I had to do it before for the post submission journey and it took a substantial amount of time.